plugins/modules/sonic_lst.py

```diff
@@ -0,0 +1,445 @@
+
+#############################################
+#                WARNING                    #
+#############################################
+#
+# This file is auto generated by the resource
+#   module builder playbook.
+#
+# Do not edit this file manually.
+#
+# Changes to this file will be over written
+#   by the resource module builder.
+#
+# Changes should be made in the model used to
+#   generate this file or in the resource module
+#   builder template.
+#
+#############################################
+
+"""
+The module file for sonic_lst
+"""
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+DOCUMENTATION = """
+---
+module: sonic_lst
+version_added: 3.1.0
+notes:
+- Tested against Enterprise SONiC Distribution by Dell Technologies.
+- Supports C(check_mode).
+short_description: Manage link state tracking (LST) configuration on SONiC
+description:
+  - This module provides configuration management of LST for devices running SONiC
+author: Shade Talabi (@stalabi1)
+options:
+  config:
+    description:
+      - LST configuration
+    type: dict
+    suboptions:
+      lst_groups:
+        description:
+          - LST groups configuration
+        type: list
+        elements: dict
+        suboptions:
+          name:
+            description:
+              - Name of LST group
+            type: str
+            required: true
+          all_evpn_es_downstream:
+            description:
+              - Indicates that the LST group tracks all EVPN ethernet segments as downstream interfaces
+            type: bool
+          all_mclags_downstream:
+            description:
+              - Indicates that the LST group tracks all MCLAGs as downstream interfaces
+            type: bool
+          group_description:
+            description:
+              - Description of LST group
+            type: str
+          group_type:
+            description:
+              - LST group type
+            type: str
+            choices: ['l3']
+          threshold_down:
+            description:
+              - Downstream ports will shut down if the threshold falls below this value
+              - Range 0-100
+            type: int
+          threshold_type:
+            description:
+              - Type of threshold calculation scheme to use
+            type: str
+            choices: ['percentage']
+          threshold_up:
+            description:
+              - Downstream ports will go online if the threshold is greater than or equal to this value
+              - Range 0-100
+            type: str
+          timeout:
+            description:
+              - Time in seconds to wait to bring up the downstream ports after the first upstream port is online
+              - Range 1-1800
+            type: int
+      interfaces:
+        description:
+          - LST configuration for interfaces
+        type: list
+        elements: dict
+        suboptions:
+          id:
```

Comment:
Since this option specifies the name of the interface, could it be changed to 'name' ?